HousingManagementSystemApi.Tests/UseCasesTests/VerifyPropertyEligibilityUseCaseTests.cs

```diff
@@ -0,0 +1,196 @@
+namespace HousingManagementSystemApi.Tests.UseCasesTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Hackney.Shared.Asset.Boundary.Response;
+    using Hackney.Shared.Asset.Domain;
+    using Hackney.Shared.Tenure.Domain;
+    using HousingManagementSystemApi.Gateways;
+    using HousingManagementSystemApi.UseCases;
+    using Microsoft.Extensions.Logging.Abstractions;
+    using Moq;
+    using Xunit;
+
+    public class VerifyPropertyEligibilityUseCaseTests
+    {
+        private readonly Mock<IAssetGateway> retrieveAssetGateway;
+        private readonly Mock<ITenureGateway> tenureGateway;
+        private readonly VerifyPropertyEligibilityUseCase sut;
+
+        public static IEnumerable<AssetType> EligibleAssetTypes = new[]
+{
+            AssetType.Flat, AssetType.House, AssetType.Dwelling,
+        };
+
+        private readonly List<TenureType> eligibleTenureTypes = new()
+        {
+            TenureTypes.Introductory,
+            TenureTypes.Secure,
+            TenureTypes.Freehold
+        };
+
+        public VerifyPropertyEligibilityUseCaseTests()
+        {
+            this.retrieveAssetGateway = new Mock<IAssetGateway>();
+            this.tenureGateway = new Mock<ITenureGateway>();
+            this.sut = new VerifyPropertyEligibilityUseCase(retrieveAssetGateway.Object, EligibleAssetTypes, tenureGateway.Object, eligibleTenureTypes, new NullLogger<VerifyPropertyEligibilityUseCase>());
+        }
+
+        [Fact]
+        public async Task GivenANullPropertyId_WhenUseCaseIsExecuted_ThenAnExceptionShouldBeThrown()
+        {
+            // Act
+            var action = async () => await this.sut.Execute(null);
+
+            // Assert
+            await Assert.ThrowsAsync<ArgumentNullException>(action);
+        }
+
+        [Fact]
+        public async Task GivenAssetCannotBeFound_WhenUseCaseIsExecuted_ThenThereWillBeAFailureResult()
+        {
+            // Arrange
+            const string HouseThatDoesntExist = "Missing";
+
+            retrieveAssetGateway.Setup(x => x.RetrieveAsset(HouseThatDoesntExist))
+                .ReturnsAsync((AssetResponseObject)null);
+
+            tenureGateway.Setup(x => x.RetrieveTenureType(It.IsAny<string>()))

```

Comment:
Is this required? I think it would error and return when it gets to the condition `if (asset == null)` no?

Reply:
This was a test left over from another file